### PR TITLE
Delete API

### PR DESCRIPTION
**Pull request by mohanraok**
@sivaprasadreddy - pls review

### CODE REVIEW

File: src/main/java/com/sivalabs/bookstore/catalog/domain/ProductService.java [hunk 1]
```diff
@@ -53,4 +53,14 @@ private TextCriteria textCriteriaMatching(String searchCriteria) {
     private Pageable pageableSortOf(int page) {
         return PageRequest.of(page, PAGE_SIZE, Sort.Direction.ASC, "name");
     }
+
+    public Optional<Product> deleteProduct(String code) {
```
Comment (sivaprasadreddy) [inline src/main/java/com/sivalabs/bookstore/catalog/domain/ProductService.java:57]: This logic can be simplified.

```
Product product = productRepository.findByCode(code).orElseThrow(()-> new ProductNotFoundException(code));
product.setDeleted(true);
productRepository.save(product);
```

I think we don't need to return the product from delete method.

**Comment by mohanraok** [comment]
> @sivaprasadreddy - addressed review comments

**Comment by rajadilipkolli** [comment]
@mohanraok , please add delete endpoint test in `ProductControllerTest.java`  as well

**Comment by mohanraok** [comment]
> @mohanraok , please add delete endpoint test in `ProductControllerTest.java` as well

Unit tests are added in ProductControllerUnitTest.java, please check.

**Review by sivaprasadreddy — APPROVED** [review]

**Pull request merged**